### PR TITLE
General update

### PR DESCRIPTION
**Pull request by nobody43**

### CODE REVIEW

File: apparmor.d/profiles-m-r/mount [hunk 1]
```diff
@@ -9,7 +9,7 @@ abi <abi/3.0>,
 include <tunables/global>
 
 @{exec_path} = /{usr/,}{s,}bin/mount
-profile mount @{exec_path} flags=(complain) {
```
Comment (nobody43) [inline apparmor.d/profiles-m-r/mount:12]: Already handled in `main.flags`?

File: apparmor.d/groups/freedesktop/pulseaudio [hunk 2]
```diff
@@ -113,10 +114,8 @@ profile pulseaudio @{exec_path} {
   /{usr/,}lib/@{multiarch}/pulse/gconf-helper mrix,
   /{usr/,}lib/pulse-*/modules/*.so mr,
 
-  /usr/share/applications/{,**} r,
   /usr/share/glib-2.0/schemas/gschemas.compiled r,
```
Comment (nobody43) [inline apparmor.d/groups/freedesktop/pulseaudio:117]: Add to `abstractions/freedesktop.org.d`?

File: apparmor.d/profiles-m-r/qbittorrent [hunk 3]
```diff
@@ -31,11 +31,13 @@ profile qbittorrent @{exec_path} {
   include <abstractions/dbus-accessibility-strict>
   include <abstractions/dbus-network-manager-strict>
   include <abstractions/dbus-gtk>
+  include <abstractions/dri-common>
   include <abstractions/dri-enumerate>
   include <abstractions/wayland>
   include <abstractions/mesa>
   include <abstractions/openssl>
   include <abstractions/ssl_certs>
+  include <abstractions/nvidia>
```
Comment (nobody43) [inline apparmor.d/profiles-m-r/qbittorrent:40]: We should start to think about something like `gpu` abstraction.

File: apparmor.d/groups/virt/virtiofsd [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# apparmor.d - Full set of apparmor profiles
+# SPDX-License-Identifier: GPL-2.0-only
+
+abi <abi/3.0>,
+
+include <tunables/global>
+
+@{LOCAL_SHARED_DIRS} = /var/lib/libvirt/shared
+include if exists <local/tunables/virtiofsd>
```
Comment (nobody43) [inline apparmor.d/groups/virt/virtiofsd:9]: This opens up the possibility to adjust local variables. Thoughts?

File: apparmor.d/groups/freedesktop/pulseaudio (hunk 2 above)
Comment (roddhjav) [reply apparmor.d/groups/freedesktop/pulseaudio:117]: Yes, I find that could be a good option.

File: apparmor.d/profiles-m-r/qbittorrent (hunk 3 above)
Comment (roddhjav) [reply apparmor.d/profiles-m-r/qbittorrent:40]: `abstractions/opencl` is pretty mucj a gpu abstraction already.

File: apparmor.d/groups/virt/virtiofsd (hunk 4 above)
Comment (roddhjav) [reply apparmor.d/groups/virt/virtiofsd:9]: Adjust by the user? no. For this you need to use [global variables](https://apparmor.pujol.io/variables/). Variables defined in a profile header are not modifiable outside.

Here, you should have something like I did in `libvirtd`: https://github.com/roddhjav/apparmor.d/blob/main/apparmor.d/groups/virt/libvirtd#L146-L150

File: apparmor.d/groups/browsers/firefox [hunk 5]
```diff
@@ -12,7 +12,7 @@ include <tunables/global>
 @{firefox_config_dirs} = @{HOME}/.mozilla/
 @{firefox_cache_dirs} = @{user_cache_dirs}/mozilla/
 
-@{exec_path} = /{usr/,}bin/@{firefox_name} @{firefox_lib_dirs}/@{firefox_name}
+@{exec_path} = /{usr/,}bin/@{firefox_name} @{firefox_lib_dirs}/@{firefox_name}{-bin,}
```
Comment (roddhjav) [inline apparmor.d/groups/browsers/firefox:15]: Edit `@{firefox_name}` instead

File: apparmor.d/groups/browsers/firefox [hunk 6]
```diff
@@ -173,6 +173,8 @@ profile firefox @{exec_path} flags=(attach_disconnected) {
   /etc/xul-ext/kwallet5.js r,
 
   owner @{HOME}/ r,
+  owner @{HOME}/@{XDG_DESKTOP_DIR}/ w,
+  owner @{HOME}/@{XDG_DOWNLOAD_DIR}/ w,
```
Comment (roddhjav) [inline apparmor.d/groups/browsers/firefox:177]: I think you should edit `abstractions/user-download-strict` instead.

File: apparmor.d/profiles-m-r/mount (hunk 1 above)
Comment (roddhjav) [reply apparmor.d/profiles-m-r/mount:12]: They should. But when removing complain flags here please double check that they are already in `main.flags`.

Comment (roddhjav) [reply apparmor.d/profiles-m-r/mount:12]: Also, you should add all your new profile in `main.flags`.

File: apparmor.d/profiles-m-r/remmina [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+  owner @{PROC}/@{pid}/task/@{tid}/comm rw,
+  owner @{PROC}/@{pid}/mountinfo r,
+
+  owner @{run}/user/@{uid}/keyring/ssh rw,
+
+  # gtk-tiny
+  /etc/gtk-3.0/settings.ini r,
+  /usr/share/themes/{,**} r,
+
+  # X-strict
```
Comment (roddhjav) [inline apparmor.d/profiles-m-r/remmina:138]: Why don't you use the X-strict abstraction?

**Comment by roddhjav** [comment]
Thanks a lot. Nice work with the dbus rules!

Regarding `loginctl`, move it to the systemd group. Also you need to add:
```
  /{usr/,}bin/less  rPx -> child-pager,
  /{usr/,}bin/more  rPx -> child-pager,
  /{usr/,}bin/pager rPx -> child-pager,
```

As well as `include <abstractions/systemd-common>` in it (this profile is part of some un-published profile I have so I did a few test with it already)

Regarding `virtiofsd`, I have already played with it a lot and I have never found proper way to profile it.

File: apparmor.d/profiles-m-r/mount (hunk 1 above)
Comment (nobody43) [reply apparmor.d/profiles-m-r/mount:12]: Double checked. Added.

File: apparmor.d/groups/freedesktop/pulseaudio (hunk 2 above)
Comment (nobody43) [reply apparmor.d/groups/freedesktop/pulseaudio:117]: Wow, I'll move it in a separate PR:
```
grep -ri '/usr/share/glib-2.0/schemas/gschemas.compiled' apparmor.d | wc -l
131
```

File: apparmor.d/profiles-m-r/qbittorrent (hunk 3 above)
Comment (nobody43) [reply apparmor.d/profiles-m-r/qbittorrent:40]: Close, but to broad for my liking.

File: apparmor.d/profiles-m-r/remmina (hunk 7 above)
Comment (nobody43) [reply apparmor.d/profiles-m-r/remmina]: That's an intermediary for the following xorg-related PR.

**Comment by nobody43** [comment]
Note: I haven't been able to test `virtiofsd` with project's `libvirtd` yet, so it might need it's following `@{user_vm_shares}` rule.

**Comment by nobody43** [comment]
I'm not a fan of whitespaces in Linux, but that's what distros already do, changed.
Not dotfile because the directory is intended for GUI access.
I'm trying not to give gross-access unless needed.

**Comment by roddhjav** [comment]
Look good. Merged thanks.

**Pull request merged**